core/src/main/java/life/qbic/identityaccess/application/user/UserRegistrationService.java

```diff
@@ -256,7 +273,14 @@ public UserExistsException() {
    * @since 1.0.0
    */
   public void confirmUserEmail(String userId) throws UserNotFoundException {
-    DomainEventPublisher.instance().subscribe(new DomainEventSubscriber<UserActivated>() {
+    DomainEventPublisher domainEventPublisher = DomainEventPublisher.instance();
+    while (!domainEventPublisher.clear()) {
+      try {
+        Thread.sleep(1);
+      } catch (InterruptedException ignored) {
+      }
```

Comment:
As I recently learned, swallowing exceptions might lead to a huge overhead when trying to figure out what happened.
(https://www.youtube.com/watch?v=LRwCE7GreSM)
In this case, I would ask that we log the error here or rethrow a RuntimeException and handle the exception on an outer level.
```suggestion
      } catch (InterruptedException ignored) {
        log.error(ignored);
      }
```
```suggestion
      } catch (InterruptedException ignored) {
        throw new RuntimeException("Unexpected interrupt.", ignored);
      }
```